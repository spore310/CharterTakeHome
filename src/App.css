.App {
  display:flex;
  flex-direction: column;
  width:100%;
  height:fit-content;
}
body{
  background: linear-gradient(to left, black 8.3%, rgb(0, 40, 60) 41.6%, rgb(0, 143, 213) 93.4%);
}