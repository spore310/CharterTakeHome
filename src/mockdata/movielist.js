export const movielist = [
    {
        "id": "SP015622350000",
        "title": "The Marksman",
        "genres": [
            "Action",
            "Thriller"
        ]
    },
    {
        "id": "SP013301360000",
        "title": "Georgetown",
        "genres": [
            "Crime drama"
        ]
    },
    {
        "id": "SP013954740000",
        "title": "The Little Things",
        "genres": [
            "Thriller",
            "Crime drama",
            "Mystery"
        ]
    },
    {
        "id": "SP015878080000",
        "title": "The Dry",
        "genres": [
            "Crime drama",
            "Mystery"
        ]
    },
    {
        "id": "SP012010760000",
        "title": "The Croods: A New Age",
        "genres": [
            "Children",
            "Comedy",
            "Adventure",
            "Animated"
        ]
    },
    {
        "id": "SP015238800000",
        "title": "The Father",
        "genres": [
            "Drama"
        ]
    },
    {
        "id": "SP015666320000",
        "title": "Land",
        "genres": [
            "Drama"
        ]
    },
    {
        "id": "SP014431440000",
        "title": "Nomadland",
        "genres": [
            "Drama"
        ]
    },
    {
        "id": "SP012496520000",
        "title": "Above Suspicion",
        "genres": [
            "Crime drama",
            "Thriller"
        ]
    },
    {
        "id": "SP014459640000",
        "title": "French Exit",
        "genres": [
            "Comedy drama"
        ]
    },
    {
        "id": "SP016143620000",
        "title": "High Ground",
        "genres": [
            "Adventure",
            "Drama"
        ]
    },
    {
        "id": "SP012813800000",
        "title": "American Fighter",
        "genres": [
            "Action"
        ]
    },
    {
        "id": "SP015653500000",
        "title": "Boogie",
        "genres": [
            "Drama"
        ]
    },
    {
        "id": "SP013109420000",
        "title": "News of the World",
        "genres": [
            "Western",
            "Adventure",
            "Drama"
        ]
    },
    {
        "id": "SP015768310000",
        "title": "Together Together",
        "genres": [
            "Comedy drama"
        ]
    },
    {
        "id": "SP014367840000",
        "title": "Judas and the Black Messiah",
        "genres": [
            "Historical drama",
            "Biography"
        ]
    },
    {
        "id": "SP015600950000",
        "title": "The Mauritanian",
        "genres": [
            "Drama",
            "Thriller"
        ]
    },
    {
        "id": "SP015997310000",
        "title": "The Resort",
        "genres": [
            "Horror",
            "Thriller"
        ]
    },
    {
        "id": "SP013723210000",
        "title": "Nobody",
        "genres": [
            "Action",
            "Thriller",
            "Comedy"
        ]
    },
    {
        "id": "SP015806650000",
        "title": "Justice Society: World War II",
        "genres": [
            "Fantasy",
            "Animated"
        ]
    },
    {
        "id": "SP016088830000",
        "title": "Percy vs Goliath",
        "genres": [
            "Drama"
        ]
    },
    {
        "id": "SP013584170000",
        "title": "Let Him Go",
        "genres": [
            "Thriller",
            "Western"
        ]
    },
    {
        "id": "SP015351890000",
        "title": "Timecrafters: The Treasure of Pirate's Cove",
        "genres": [
            "Adventure",
            "Fantasy"
        ]
    },
    {
        "id": "SP016070160000",
        "title": "The Virtuoso",
        "genres": [
            "Thriller"
        ]
    },
    {
        "id": "SP015413280000",
        "title": "The Killing of Two Lovers",
        "genres": [
            "Drama"
        ]
    },
    {
        "id": "SP013727990000",
        "title": "Riders of Justice",
        "genres": [
            "Action",
            "Thriller",
            "Dark comedy"
        ]
    },
    {
        "id": "SP015893830000",
        "title": "616 Wilford Lane",
        "genres": [
            "Horror"
        ]
    },
    {
        "id": "SP010286020000",
        "title": "Locked In",
        "genres": [
            "Thriller"
        ]
    },
    {
        "id": "SP013515940000",
        "title": "Promising Young Woman",
        "genres": [
            "Dark comedy",
            "Thriller"
        ]
    },
    {
        "id": "SP015766610000",
        "title": "In the Earth",
        "genres": [
            "Horror",
            "Thriller"
        ]
    },
    {
        "id": "SP016282990000",
        "title": "The Retreat",
        "genres": [
            "Thriller"
        ]
    },
    {
        "id": "SP013848680000",
        "title": "North Hollywood",
        "genres": [
            "Comedy drama"
        ]
    },
    {
        "id": "SP016055960000",
        "title": "The Djinn",
        "genres": [
            "Horror",
            "Thriller"
        ]
    },
    {
        "id": "SP015997800000",
        "title": "Trigger Point",
        "genres": [
            "Thriller",
            "Action"
        ]
    },
    {
        "id": "SP016010220000",
        "title": "2149: The Aftermath",
        "genres": [
            "Science fiction"
        ]
    }
]