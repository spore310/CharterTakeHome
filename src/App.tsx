import React from 'react';
import './App.css';
import { InputField } from './components/InputField/inputField';
import List from './components/searchResult'
import InfoCard from './components/InfoCard';
import style from './app.module.css';
function App() {
  
  return (
    <div className="App">
      <InputField/>
      <div className={style.container}> 
        <List/>
        <InfoCard/>
      </div>
    </div>
  );
}

export default App;
