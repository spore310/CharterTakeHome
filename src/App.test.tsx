import React from 'react';
import { fireEvent, render, screen } from '@testing-library/react';
import { Provider } from 'react-redux';

import renderObj from './utils/test-utils';
import App from './App';
import { movielist } from './mockdata/movielist';

describe("InputField", () => {

    test("snapshot test", () => {
        const { renderWithProvider } = renderObj;
        const view = renderWithProvider(<App />);
        expect(view.asFragment()).toMatchSnapshot();
    });
/* 

*/
    test("search input should update the search list",()=>{
        const { renderWithProvider } = renderObj;
       renderWithProvider(<App />, {
            preloadedState: {
                movie: {
                    movieList: movielist,
                    genres: ['Action','Thriller','Crime drama'],
                    loading: false,
                    searchList: [],
                    keyword: '',
                    genre: '',
                    infoSelect: { id: "0", title: '', genres: [''] }
                }
            }
        });
        const inputEl = screen.getByTestId("inputfield-input");
        expect(inputEl).toBeInTheDocument()
        fireEvent.change(inputEl, {target:{value:"geo"}});
        let movieItemCardEl = screen.getByText("Georgetown");
        expect(movieItemCardEl).toBeInTheDocument();
        fireEvent.change(inputEl, {target:{value:"geg"}});
        movieItemCardEl = screen.queryByText("Georgetown") as any;
        expect(movieItemCardEl).not.toBeInTheDocument();

    })
})