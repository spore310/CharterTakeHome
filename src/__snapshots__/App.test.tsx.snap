// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputField snapshot test 1`] = `
<DocumentFragment>
  <div
    class="App"
  >
    <div
      class="wrapper"
      data-testid="inputfield-wrapper"
    >
      <div
        class="inputWrapper"
      >
        <input
          data-testid="inputfield-input"
          type="text"
          value=""
        />
        <div
          class="selectorWrapper"
        >
          <button
            class="genreButton"
          >
            Genre
          </button>
          <div
            class="auto"
          />
        </div>
      </div>
    </div>
    <div
      class="container"
    >
      <div
        class="container"
      >
        <ul
          class="list"
        >
          <div
            class="noResults"
          >
            No Results
          </div>
        </ul>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
